docs/modules/ROOT/pages/api/erc721.adoc

```diff
@@ -0,0 +1,654 @@
+
+See <<IERC721-safe_transfer_from,IERC721::safe_transfer_from>>.
+
+Requirements:
+
+- Caller is either approved or the `token_id` owner.
+- `to` is not the zero address.
+- `from` is not the zero address.
+- `token_id` exists.
+- `to` is either an account contract or supports the <<IERC721Receiver,IERC721Receiver>> interface.
+
+[.contract-item]
+[[ERC721-transfer_from]]
+==== `[.contract-item-name]#++transfer_from++#++(ref self: ContractState, from: ContractAddress, to: ContractAddress, token_id: u256)++` [.item-kind]#external#
+
+See <<IERC721-transfer_from,IERC721::transfer_from>>.
+
+Requirements:
+
+- Caller either approved or the `token_id` owner.
+- `to` is not the zero address.
+- `from` is not the zero address.
+- `token_id` exists.
+
+[.contract-item]
+[[ERC721-approve]]
+==== `[.contract-item-name]#++approve++#++(ref self: ContractState, to: ContractAddress, token_id: u256)++` [.item-kind]#external#
+
+See <<IERC721-approve,IERC721::approve>>.
+
+Requirements:
+
+- The caller is either an approved operator or the `token_id` owner.
+- `to` cannot be the token owner or the zero address.
+- `token_id` exists.
+
+[.contract-item]
+[[ERC721-set_approval_for_all]]
+==== `[.contract-item-name]#++set_approval_for_all++#++(ref self: ContractState, operator: ContractAddress, approved: bool)++` [.item-kind]#external#
+
+See <<IERC721-set_approval_for_all,IERC721::set_approval_for_all>>.
+
+Requirements:
+
+- `operator` cannot be the caller.
+
+[.contract-item]
+[[ERC721-get_approved]]
+==== `[.contract-item-name]#++get_approved++#++(self: @ContractState, token_id: u256) -> u256++` [.item-kind]#external#
+
+See <<IERC721-get_approved,IERC721::get_approved>>.
+
+Requirements:
+
+- `token_id` exists.
+
+[.contract-item]
+[[ERC721-is_approved_for_all]]
+==== `[.contract-item-name]#++is_approved_for_all++#++(self: @ContractState, owner: ContractAddress, operator: ContractAddress) -> bool++` [.item-kind]#external#
+
+See <<IERC721-is_approved_for_all,IERC721::is_approved_for_all>>.
+
+[.contract-item]
+[[ERC721-name]]
+==== `[.contract-item-name]#++name++#++(self: @ContractState) -> felt252++` [.item-kind]#external#
+
+See <<IERC721Metadata-name,IERC721Metadata::name>>.
+
+[.contract-item]
+[[ERC721-symbol]]
+==== `[.contract-item-name]#++symbol++#++(self: @ContractState) -> felt252++` [.item-kind]#external#
+
+See <<IERC721Metadata-symbol,IERC721Metadata::symbol>>.
+
+[.contract-item]
+[[ERC721-token_uri]]
+==== `[.contract-item-name]#++token_uri++#++(self: @ContractState, token_id: u256) -> felt252++` [.item-kind]#external#
+
+See <<IERC721Metadata-token_uri,IERC721Metadata::token_uri>>.
+
+[.contract-item]
+[[ERC721-balanceOf]]
+==== `[.contract-item-name]#++balanceOf++#++(self: @ContractState, account: ContractAddress) -> u256++` [.item-kind]#external#
+
+See <<IERC721-balance_of,IERC721::balance_of>>.
+
+Supports the Cairo v0 convention of writing external methods in camelCase as discussed {casing-discussion}.
```

Comment:
I don't think there's much value in repeating this on every camelCase method. I vote for removing them.